### ISSUE TITLE
Disable http/2 and use http/1.1

### ISSUE THREAD

**Issue by CristianIonutIonescu** [embedthis/esp#88]
Hello again!

First part:
So..after updating from esp 7.0.2 to 8.0.0 I came across another problem: when configuring my HTTPS connection to the web server, I cannot access the applications pages. I changed nothing to my source code between esp 7.1.0 and esp 8.0.0. Are there any configuration changes that are to be made in order to use https?

For replicating the issue, one can build and run any sample that uses https (login-auth for example) and see the problem. (ubuntu 16.04, default build)

My guess is that the issue is due to switching to http2 so I tried to disable it and I encountered the second problem:

When trying to build esp with http2 false, I encounter a build error:

### Environment
ESP version: 8.0.0 build from source
O/S: Linux Ubuntu 16.04
Hardware PC
Customization: http2 disabled(attempting at least :) )

#### Custom.me file:
```
Me.load({
    settings: {
        http: {
            http2: false,
        },
	}
})
```

### Build steps:

```
$ ./configure
$ me
```

### Error message

```
      [Copy] build/linux-x64-debug/inc/edi.h
      [Copy] build/linux-x64-debug/inc/esp.h
      [Copy] build/linux-x64-debug/inc/mdb.h
      [Copy] build/linux-x64-debug/inc/sqlite3.h
   [Compile] build/linux-x64-debug/obj/sqlite3.o
      [Link] build/linux-x64-debug/bin/libsql.so
      [Copy] build/linux-x64-debug/inc/osdep.h
      [Copy] build/linux-x64-debug/inc/embedtls.h
      [Copy] build/linux-x64-debug/inc/mbedtls-config.h
      [Copy] build/linux-x64-debug/inc/mbedtls.h
   [Compile] build/linux-x64-debug/obj/mbedtls.o
   [Archive] build/linux-x64-debug/bin/libmbedtls.a
      [Copy] build/linux-x64-debug/inc/mpr.h
   [Compile] build/linux-x64-debug/obj/mpr-mbedtls.o
   [Archive] build/linux-x64-debug/bin/libmpr-mbedtls.a
   [Compile] build/linux-x64-debug/obj/mprLib.o
      [Link] build/linux-x64-debug/bin/libmpr.so
      [Copy] build/linux-x64-debug/inc/pcre.h
   [Compile] build/linux-x64-debug/obj/pcre.o
      [Link] build/linux-x64-debug/bin/libpcre.so
      [Copy] build/linux-x64-debug/inc/http.h
   [Compile] build/linux-x64-debug/obj/httpLib.o
me: ERROR: Building target /home/user/Downloads/esp-8.0.0/build/linux-x64-debug/obj/httpLib.o

Command failure: src/http/httpLib.c: In function ‘parseLimitsConnections’:
src/http/httpLib.c:4437:18: error: ‘HttpLimits {aka struct HttpLimits}’ has no member named ‘streamsMax’
     route->limits->streamsMax = httpGetInt(prop->value);
                  ^
src/http/httpLib.c: In function ‘httpSetQueueLimits’:
src/http/httpLib.c:16767:24: error: ‘HttpLimits {aka struct HttpLimits}’ has no member named ‘window’
         window = limits->window;
                        ^
src/http/httpLib.c: In function ‘httpCreateStream’:
src/http/httpLib.c:22108:52: error: ‘HttpLimits {aka struct HttpLimits}’ has no member named ‘txStreamsMax’
     if (!peerCreated && ((net->ownStreams >= limits->txStreamsMax) || (net->ownStreams >= limits->streamsMax))) {
                                                    ^
src/http/httpLib.c:22108:97: error: ‘HttpLimits {aka struct HttpLimits}’ has no member named ‘streamsMax’
     if (!peerCreated && ((net->ownStreams >= limits->txStreamsMax) || (net->ownStreams >= limits->streamsMax))) {
                                                                                                 ^
src/http/httpLib.c:22110:19: error: ‘HttpLimits {aka struct HttpLimits}’ has no member named ‘txStreamsMax’
             limits->txStreamsMax, limits->streamsMax);
                   ^
src/http/httpLib.c:22110:41: error: ‘HttpLimits {aka struct HttpLimits}’ has no member named ‘streamsMax’
             limits->txStreamsMax, limits->streamsMax);
```
Did I miss to add something to custom.me file or this is a bug?
Is mandatory to use http2 for esp? If yes, can you provide an example/sample with what configurations are needed to communicate between browser and esp over http2?

If you need any other information, please let me know and I will provide them as soon as possible.

Thank you,
Cristian Ionescu

**Comment by mobrien** [comment]
Thank you again for the report.

Several issues here, all fixed and checked in to 8.0.1 dev or master.

Should now build without HTTP/2 and should work with HTTP/2 with browser requests.

A late fix for some  compiler warnings broke some browser requests. All our unit tests passed, so we'll have to expand them to include the failing requests.

Please test it and let me know.

Thanks

**Comment by dev-embedthis** [comment]
Fixed in the master branch.
